src/lib/APS2.cpp

```diff
@@ -1513,7 +1457,37 @@ void APS2::setup_VCXO() {
   write_SPI(msg);
 }
 
-void APS2::setup_DAC(const int &dac)
+void APS2::align_DAC_clock(int dac) {
+  // toggle DAC clocks until DATACLK_OUT comes up "aligned" to system 600
+  const vector<uint32_t> PHASE_COUNT_ADDR = {PHASE_COUNT_A_ADDR,
+                                             PHASE_COUNT_B_ADDR};
+  // Loop over number of tries
+  for (int ct = 0; ct < MAX_DAC_CLOCK_PHASE_TEST_TRIES; ct++) {
+    // register returns a value in [0, 0xffff]. Re-interpret as portion of
+    // circle
+    auto dac_clk_phase =
+        static_cast<double>(read_memory(PHASE_COUNT_ADDR[dac], 1)[0]) /
+        (0xffff);
+
+    FILE_LOG(logDEBUG1) << ipAddr_ << " measured DAC "
+                        << ((dac == 0) ? "A" : "B") << " clock phase of "
+                        << dac_clk_phase;
+
+    if (dac_clk_phase < 0.5) {
+      // done with this channel
+      return;
+    } else {
+      // toggle DAC clock to try and get different phase
+      toggle_DAC_clock(dac);
+      // wait for phase count to run
+      std::this_thread::sleep_for(std::chrono::milliseconds(1));
+    }
+  }
+  // if we got here then we never aligned
+  FILE_LOG(logERROR) << ipAddr_ << " failed to align DAC " << dac;
```

Comment:
Should this throw an error so that we can report an error at the application level?